compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -272,6 +276,8 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
             checkNoConstructorProxy(tree)
             transformSelect(tree, Nil)
         case tree: Apply =>
+          if tree.symbol.is(Inline) && !Inliner.inInlineMethod then
```

Comment:
Why is this needed here? Don't we get to an Ident or Select eventually, which sets `needsInlining`. Also, why do we handle `Apply` here but not `TypeApply`. If this is in fact needed for some reason it would be good to add a comment.

Reply:
It is redundant. I removed it.